src/renate/memory/buffer.py

```diff
@@ -231,25 +258,28 @@ def __init__(
         super().__init__(
             storage_mode=storage_mode, transform=transform, target_transform=target_transform
         )
-        self._max_size = 1
+        self._max_size = 16
 
     def _update(self, dataset: Dataset, metadata: DataDict) -> None:
         for i in range(len(dataset)):
             self._append(dataset[i], {key: value[i] for key, value in metadata.items()})
 
-    def _append(self, data: DataTuple, metadata: DataDict) -> None:
+    def _append(self, data: NestedTensors, metadata: NestedTensors) -> None:
         """Appends a data point to the internal storage.
 
         Initializes a new buffer with twice the size if the current one is full.
         """
         super()._append(data, metadata)
         if self._size == self._max_size:
-            data_containers = [self._data_points, metadata]
-            for data_container in data_containers:
-                for key in data_container:
-                    data_container[key] = torch.cat(
-                        [data_container[key], torch.empty_like(data_container[key])], dim=0
-                    )
+            # Extend the size of the storage by a factor of 2.
+            current_data_points = self._data_points
+            self._data_points = _make_storage(
+                _get_data_point(current_data_points, 0), self._max_size * 2
+            )
+            _insert_data_point(self._data_points, slice(self._max_size), current_data_points)
+            current_metadata = self.metadata
+            self.metadata = _make_storage(_get_data_point(current_metadata, 0), self._max_size * 2)
+            _insert_data_point(self.metadata, slice(self._max_size), current_metadata)
             self._max_size *= 2
```

Comment:
nit: would update `_max_size` first and then use it to create the new storage.

Reply:
I don't really like that since I need the "old" `max_size` in the `_insert_data_point`.